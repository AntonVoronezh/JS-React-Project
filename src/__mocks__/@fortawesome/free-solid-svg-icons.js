const Icon = {
	prefix: 'prefix',
	iconName: 'iconName',
	icon: [
		//   width,
		//   height,
		//   ligatures,
		//   unicode,
		//   svgPathData
	],
};

export { Icon };
