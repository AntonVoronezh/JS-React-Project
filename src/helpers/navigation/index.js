export { routes } from './routes';
export { navLinks, Navigation } from './navigation';
