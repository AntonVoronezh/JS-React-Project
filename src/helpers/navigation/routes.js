import {
	MainPage,
	NotFoundPage,
	ModelsPage,
	ModelsListPage,
	ModelEditPage,
	ModelCreatePage,
	FieldsPage,
	FieldCreatePage,
	FieldsListPage,
	FieldEditPage
} from '../../containers/pages';
import { path } from '../constants';

const routes = [
	{
		path: '/',
		exact: true,
		component: MainPage,
	},
	{
		path: `/${path.models}`,
		component: ModelsPage,
		routes: [
			{
				path: `/${path.models}/${path.list}`,
				component: ModelsListPage,
			},
			{
				path: `/${path.models}/${path.edit}/:${path.id}`,
				component: ModelEditPage,
			},
			{
				path: `/${path.models}/${path.create}`,
				component: ModelCreatePage,
			},
		],
	},
	{
		path: `/${path.fields}`,
		component: FieldsPage,
		routes: [
			{
				path: `/${path.fields}/${path.list}`,
				component: FieldsListPage,
			},
			{
				path: `/${path.fields}/${path.edit}/:${path.id}`,
				component: FieldEditPage,
			},
			{
				path: `/${path.fields}/${path.create}`,
				component: FieldCreatePage,
			},
		],
	},
	{
		path: '',
		component: NotFoundPage,
	},
];

export { routes };
