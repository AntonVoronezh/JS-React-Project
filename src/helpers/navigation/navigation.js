import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChartLine, faBiohazard, faBezierCurve, faAlignJustify, faBroom } from '@fortawesome/free-solid-svg-icons';
import { NavLink } from 'react-router-dom';
import { Trans } from 'react-i18next';

import { path } from '../constants';

const navLinks = {
	mainNav: [
		{
			to: '/',
			title: 'navigation.main',
			icon: faChartLine,
		},
		{
			to: `/${path.models}`,
			title: 'navigation.models.title',
			icon: faBiohazard,
		},
		{
			to: `/${path.fields}`,
			title: 'navigation.attrubutes.title',
			icon: faBezierCurve,
		},
	],
	modelNav: [
		{
			to: `/${path.models}/${path.list}`,
			title: 'navigation.models.list',
			icon: faAlignJustify,
		},
		{
			to: `/${path.models}/${path.create}`,
			title: 'navigation.models.create',
			icon: faBroom,
		},
	],
	fieldNav: [
		{
			to: `/${path.fields}/${path.list}`,
			title: 'navigation.attrubutes.list',
			icon: faAlignJustify,
		},
		{
			to: `/${path.fields}/${path.create}`,
			title: 'navigation.attrubutes.create',
			icon: faBroom,
		},
	],
};

const Navigation = ({ arrLinks }) => {
	return (
		<ul className="navbar-nav">
			{arrLinks.map(({ title, to, icon }) => {
				return (
					<li key={to} className="nav-item">
						<NavLink to={to} className="nav-item  nav-link  header-nav" activeClassName="active2">
							<div className="nav-wrap">
								<FontAwesomeIcon icon={icon} className="mr-1" />
								<Trans i18nKey={title} />
							</div>
						</NavLink>
					</li>
				);
			})}
		</ul>
	);
};

export { navLinks, Navigation };
