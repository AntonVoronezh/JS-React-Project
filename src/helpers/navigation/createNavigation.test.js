import React from 'react';
import { render, shallow } from 'enzyme';
import { Icon } from '@fortawesome/free-solid-svg-icons';

import { Navigation } from './navigation';

describe('function createNavigation', () => {
	const mockLinks = {
		mainNav: [
			{
				to: '/',
				title: 'navigation.main',
				icon: Icon,
			},
		],
	};

	it('should be one nav', () => {
		let wrapper = shallow(<Navigation arrLinks={mockLinks.mainNav} />);
		expect(wrapper).toMatchSnapshot();
	});
});
