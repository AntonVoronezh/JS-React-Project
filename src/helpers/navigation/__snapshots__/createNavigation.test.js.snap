// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function createNavigation should be one nav 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    arrLinks={
      Array [
        Object {
          "icon": Object {
            "icon": Array [],
            "iconName": "iconName",
            "prefix": "prefix",
          },
          "title": "navigation.main",
          "to": "/",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active2"
            className="nav-item  nav-link  header-nav"
            to="/"
          >
            <div
              className="nav-wrap"
            >
              <FontAwesomeIcon
                className="mr-1"
                icon={
                  Object {
                    "icon": Array [],
                    "iconName": "iconName",
                    "prefix": "prefix",
                  }
                }
              />
              <Trans
                i18nKey="navigation.main"
              />
            </div>
          </NavLink>
        </li>,
      ],
      "className": "navbar-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "/",
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active2"
            className="nav-item  nav-link  header-nav"
            to="/"
          >
            <div
              className="nav-wrap"
            >
              <FontAwesomeIcon
                className="mr-1"
                icon={
                  Object {
                    "icon": Array [],
                    "iconName": "iconName",
                    "prefix": "prefix",
                  }
                }
              />
              <Trans
                i18nKey="navigation.main"
              />
            </div>
          </NavLink>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active2",
            "children": <div
              className="nav-wrap"
            >
              <FontAwesomeIcon
                className="mr-1"
                icon={
                  Object {
                    "icon": Array [],
                    "iconName": "iconName",
                    "prefix": "prefix",
                  }
                }
              />
              <Trans
                i18nKey="navigation.main"
              />
            </div>,
            "className": "nav-item  nav-link  header-nav",
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  className="mr-1"
                  icon={
                    Object {
                      "icon": Array [],
                      "iconName": "iconName",
                      "prefix": "prefix",
                    }
                  }
                />,
                <Trans
                  i18nKey="navigation.main"
                />,
              ],
              "className": "nav-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "mr-1",
                  "icon": Object {
                    "icon": Array [],
                    "iconName": "iconName",
                    "prefix": "prefix",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "i18nKey": "navigation.main",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-item"
          >
            <NavLink
              activeClassName="active2"
              className="nav-item  nav-link  header-nav"
              to="/"
            >
              <div
                className="nav-wrap"
              >
                <FontAwesomeIcon
                  className="mr-1"
                  icon={
                    Object {
                      "icon": Array [],
                      "iconName": "iconName",
                      "prefix": "prefix",
                    }
                  }
                />
                <Trans
                  i18nKey="navigation.main"
                />
              </div>
            </NavLink>
          </li>,
        ],
        "className": "navbar-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "/",
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active2"
              className="nav-item  nav-link  header-nav"
              to="/"
            >
              <div
                className="nav-wrap"
              >
                <FontAwesomeIcon
                  className="mr-1"
                  icon={
                    Object {
                      "icon": Array [],
                      "iconName": "iconName",
                      "prefix": "prefix",
                    }
                  }
                />
                <Trans
                  i18nKey="navigation.main"
                />
              </div>
            </NavLink>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active2",
              "children": <div
                className="nav-wrap"
              >
                <FontAwesomeIcon
                  className="mr-1"
                  icon={
                    Object {
                      "icon": Array [],
                      "iconName": "iconName",
                      "prefix": "prefix",
                    }
                  }
                />
                <Trans
                  i18nKey="navigation.main"
                />
              </div>,
              "className": "nav-item  nav-link  header-nav",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    className="mr-1"
                    icon={
                      Object {
                        "icon": Array [],
                        "iconName": "iconName",
                        "prefix": "prefix",
                      }
                    }
                  />,
                  <Trans
                    i18nKey="navigation.main"
                  />,
                ],
                "className": "nav-wrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "mr-1",
                    "icon": Object {
                      "icon": Array [],
                      "iconName": "iconName",
                      "prefix": "prefix",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "i18nKey": "navigation.main",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
